central/notifiers/microsoftsentinel/sentinel.go

```diff
@@ -70,58 +72,75 @@ func (s sentinel) ProtoNotifier() *storage.Notifier {
 }
 
 func (s sentinel) Test(ctx context.Context) *notifiers.NotifierError {
-	// TODO(ROX-25857): test call will be updated when implementing the table in Microsoft Sentinel
+	alert := s.getTestAlert()
+
+	err := s.AlertNotify(ctx, alert)
+	if err != nil {
+		return notifiers.NewNotifierError("could not send event", err)
+	}
+	return nil
+}
+
+func (s sentinel) getTestAlert() *storage.Alert {
 	alert := &storage.Alert{
-		ClusterId:   "cluster-id",
-		ClusterName: "cluster-01",
-		Namespace:   "default",
-		NamespaceId: "default-ns-id",
 		Policy: &storage.Policy{
-			Id:              "policy-id",
-			Name:            "some-policy",
-			Categories:      make([]string, 0),
-			Description:     "",
-			Remediation:     "",
-			LifecycleStages: []storage.LifecycleStage{storage.LifecycleStage_BUILD, storage.LifecycleStage_DEPLOY},
+			Name:        "test-policy",
+			Description: "Test policy description",
 		},
-		Time:          timestamppb.Now(),
-		FirstOccurred: timestamppb.Now(),
-		ResolvedAt:    timestamppb.Now(),
-		State:         storage.ViolationState_ACTIVE,
-		SnoozeTill:    timestamppb.Now(),
-	}
-	if err := s.AlertNotify(ctx, alert); err != nil {
-		return notifiers.NewNotifierError("could not send event", err)
+		ClusterName: "test-cluster",
+		ClusterId:   uuid.NewDummy().String(),
+		Namespace:   "test-namespace",
 	}
-	return nil
+	return alert
 }
 
 func (s sentinel) AlertNotify(ctx context.Context, alert *storage.Alert) error {
 	if !features.MicrosoftSentinelNotifier.Enabled() {
 		return errors.New("Microsoft Sentinel notifier is disabled.")
 	}
 
+	bytesToSend, err := s.prepareLogsToSend(alert)
+	if err != nil {
+		return err
+	}
+
+	err = retry.WithRetry(func() error {
```

Comment:
Should we determine a set of retryable vs not retryable errors?  The splunk notifier contains an example of that concept.